### PR TITLE
Use Topkg

### PR DESCRIPTION
**Pull request by verbosemode**
* Switching from Oasis to Topkg
* I got rid of lib_test/Makefile. topkg build && topkg test can now be used to run the tests
* topkg lint complains about opam/_tags being inconsistent, but this is mainly because of ounit, having different names in ocamlfind and opam
* I made top/ipaddr_top.ml 's build depend on unix. I hope this assumption is correct
* I had to create top/ipaddr_top.mli. Without it, ocamlbuild was complaining about a missing ipaddr_top.mly file 

I've tested it in a new 4.03.0 switch with a local pin, but please have a closer look at the PR. I'm not that expericed with all the OCaml build tools yet.

### CODE REVIEW

File: _tags [hunk 1]
```diff
@@ -1,35 +1,11 @@
...
-# OASIS_STOP
-true: debug, principal, safe_string
+true: bin_annot, principal, safe_string
+true: package(bytes sexplib ppx_sexp_conv)
 
+"lib" : include
+
+<lib/ipaddr_unix.*> : package(unix)
+
+"top" : include
```
Comment (hannesm) [inline _tags]: I'd guess this is not needed

File: _tags [hunk 2]
```diff
@@ -1,35 +1,11 @@
...
+
+"lib_test" : include
```
Comment (hannesm) [inline _tags]: same, don't think this is needed

File: opam [hunk 3]
```diff
@@ -16,24 +16,19 @@ tags: [
...
-]
-install: [
-  [make "install"]
-]
-remove: [
-  ["ocamlfind" "remove" "ipaddr"]
+  [ "ocaml" "pkg/pkg.ml" "build" "--pinned" "%{pinned}%"
+    "--with-base-unix" "%{base-unix:installed}%"
+    "--with-top" "%{base-unix:installed}%" ]
 ]
```
Comment (hannesm) [inline opam]: if you re-add a `build-test` rule, the travis CI will also compile, run, and execute those :)

```
build-test: [
  [ "ocaml" "pkg/pkg.ml" "build" "--pinned" "%{pinned}%" "--tests" "true" ]
  [ "ocaml" "pkg/pkg.ml" "test" ]
]
```

**Comment by hannesm** [comment]
The `ounit` vs `oUnit` can be ignored in pkg/pkg.ml by (as done in [functoria](
https://github.com/mirage/functoria/blob/d955e45257dd49f3c6ade5eb7464e67a3776dca7/pkg/pkg.ml#L6-L11):

```OCaml
let opams =
  let lint_deps_excluding = Some ["ounit"; "oUnit"] in
  [Pkg.opam_file ~lint_deps_excluding "opam"]

let () =
Pkg.describe ~opams "functoria" @@ fun c ->
```

Really nice done!  I commented on some minor things

**Comment by verbosemode** [comment]
Thanks for your comments. I've incorporated the changes and tested them.

It seems like I've broken some of your nice source code comments while rebasing and fiddling with the git commits.

File: opam [hunk 4]
```diff
@@ -16,24 +16,22 @@ tags: [
...
 
 build: [
-  ["ocaml" "setup.ml" "-configure" "--prefix" prefix "--%{ounit:enable}%-tests"]
-  [make "build"]
+  [ "ocaml" "pkg/pkg.ml" "build" "--pinned" "%{pinned}%"
+    "--with-base-unix" "%{base-unix:installed}%"
+    "--with-top" "%{base-unix:installed}%" ]
```
Comment (hannesm) [inline opam]: since `top` is true, whenever `base-unix` is true, you can get rid of top...

**Comment by verbosemode** [comment]
Good catch. We'll get there :-)

**Comment by hannesm** [comment]
LGTM! not sure whose code this library is, likely @dsheets should ack this PR (via the contributors statistics on GitHub)

**Comment by hannesm** [comment]
@dsheets will you merge + make a minor release? (or @avsm if he feels like?)

**Comment by dsheets** [comment]
Ok, I will release.

**Comment by dsheets** [comment]
ocaml/opam-repository#7833

**Pull request merged**